cmd/thv-operator/pkg/httpclient/client.go

```diff
@@ -0,0 +1,76 @@
+// Package httpclient provides HTTP client functionality for API operations
+package httpclient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+const (
+	// DefaultTimeout is the default timeout for HTTP requests
+	DefaultTimeout = 30 * time.Second
+
+	// UserAgent is the user agent string for HTTP requests
+	UserAgent = "toolhive-operator/1.0"
+)
+
+// Client is an interface for HTTP operations
+type Client interface {
+	// Get performs an HTTP GET request and returns the response body
+	Get(ctx context.Context, url string) ([]byte, error)
+}
+
+// DefaultClient is the default HTTP client implementation
+type DefaultClient struct {
+	client  *http.Client
+	timeout time.Duration
+}
+
+// NewDefaultClient creates a new default HTTP client with the specified timeout
+// If timeout is 0, uses DefaultTimeout
+func NewDefaultClient(timeout time.Duration) Client {
+	if timeout == 0 {
+		timeout = DefaultTimeout
+	}
+	return &DefaultClient{
```

Comment:
Should the default client use TLS? There is a `NewHttpClientBuilder` in the codebase that could be used for this..